### DISCUSSION TITLE
Improvements to exception serialization?

### DISCUSSION THREAD

**Discussion by darren-clark** [Azure/azure-functions-durable-extension#2098] [Ideas]
I'm getting into a place where I'm throwing exception that the default serializer won't handle. 
In my case it's pretty easy thing to handle with a ```JsonConverter```.
However that means I need to replace the ```JsonSerializerOptions``` by injecting an ```IErrorSerializerSettingsFactory```

All I really want to do is just add a JsonConverter to the serializer. The default(and internal) serializer uses a custom(also internal) contract resolver, which looks like it's probably useful.

At this point I'm trying to do one of two things:
1) Replace the DI registration of ```IErrorSerializerSettingsFactory``` with one that wraps the current one. Which is unpleasant because I then need to have some sort of decorator implementation or inject IServiceProvider into my implementation, call ```GetServices``` and then find the "not me" one. Made more ugly that the default implementation also caches the settings, which is good, but also means that I don't _always_ want to add my converter.
2) Take advantage of the caching to find someplace where I can execute code once after the DI container is build(I've been trying to do this for approx 3 hours now), and add it just once there.

My suggestion would be to support registration of a new interface ```IErrorSerializerSettingsConfiguration``` or something that the default factory would enumerate implementations of and pass the settings through them sequentially before caching it.

**Comment by cgillum** [comment]
Thanks @darren-clark for this idea! I went ahead and converted it into an issue and put it on the backlog. https://github.com/Azure/azure-functions-durable-extension/issues/2103
